src/operations.rs

```diff
@@ -858,6 +862,10 @@ pub fn create_symlink(
     let updating = _remove_symlink(&symlink_path)?;
 
     match channel {
```

Comment:
This match is getting a bit unwieldy. Perhaps each match could be routed to a dedicated function, rather than having the code inlined